File: Capture the Flag/src/main/java/me/yoursole/ctf/Commands/Top.java
package me.yoursole.ctf.Commands;

import org.bukkit.ChatColor;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.HashMap;

public class Top implements CommandExecutor {
    public HashMap<String, Long> cooldowns = new HashMap<>();
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        int cooldownTime = 20;
        if(cooldowns.containsKey(sender.getName())) {
            long secondsLeft = ((cooldowns.get(sender.getName())/1000)+cooldownTime) - (System.currentTimeMillis()/1000);
            if(secondsLeft>0) {
                sender.sendMessage(ChatColor.RED+"You cant use /top for another "+ secondsLeft +((secondsLeft!=1)?" seconds!":"second!"));
                return true;
            }
        }


        cooldowns.put(sender.getName(), System.currentTimeMillis());
        Player player = (Player) sender;
        World world = player.getWorld();
        Location highest = world.getHighestBlockAt(player.getLocation()).getLocation();
        highest.setDirection(player.getLocation().getDirection());
        highest.add(0,1,0);
        player.teleport(highest);





        return true;
    }
}

File: src/main/kotlin/me/yoursole/ctf/commands/Top.kt
package me.yoursole.ctf.commands

import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands.*
import net.minecraft.server.level.ServerPlayer
import java.util.UUID


object Top : BrigadierCommand {
    private val cooldowns = HashMap<UUID, Long>()
    private val cooldownTime = 20

    override fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(literal("top").requires { it.entity is ServerPlayer }.executes { ctx ->
            val player = ctx.source.playerOrException.bukkitEntity
            if (cooldowns.containsKey(player.uniqueId)) {
                val secondsLeft = cooldowns[player.uniqueId]!! / 1000 + cooldownTime - System.currentTimeMillis() / 1000
                if (secondsLeft > 0) {
                    player.sendMessage("§aYou can't use /top for another $secondsLeft ${if (secondsLeft != 1L) "seconds!" else "second!"}")
                    return@executes Command.SINGLE_SUCCESS
                }
            }
            cooldowns[player.uniqueId] = System.currentTimeMillis()
            val world = player.world
            val highest = world.getHighestBlockAt(player.location).location
            highest.direction = player.location.direction
            highest.add(0.0, 1.0, 0.0)
            player.teleport(highest)
            player.sendMessage("§aWhoosh!")
            Command.SINGLE_SUCCESS
        })
    }
}